python/ray/actor.py

```diff
@@ -337,8 +352,19 @@ def register_actor_signatures(worker, driver_id, class_name,
                                                        actor_method_name).id()
         worker.function_properties[driver_id][function_id] = (
             # The extra return value is an actor dummy object.
+            # In the cases wher actor_method_cpus is None, that value should
+            # never be used.
             FunctionProperties(num_return_vals=num_return_vals + 1,
-                               resources={"CPU": 1},
+                               resources={"CPU": actor_method_cpus},
+                               max_calls=0))
+
+    if actor_creation_resources is not None:
```

Comment:
Hm.. some of this needs to be rethought, but basically, `submit_task` looks up the task resource requirements based on the info in `worker.function_properties`, so we need to fill that out so that `submit_task` does the right thing. But maybe the right thing is for resources to be passed into `submit_task` instead.

Basically, whenever a worker might submit an actor creation task, then `register_actor_signatures ` needs to have been called with `actor_creation_resources` not equal to `None`.

Reply:
Hmm, are there cases where it is equal to `None` then?

Reply:
When a worker turns into an actor, it calls `register_actor_signatures`, and the `actor_creation_resources` data isn't really available at that point. I could pass it in by storing it in redis or through some other mechanism, but it seemed unnecessary. If this sounds a little strange, part of the issue is that there is a difference in the data that is needed at function invocation time versus at function execution time, but we're using `worker.function_properties` to store both things.